Sources/Foundation/RunLoop.swift

```diff
@@ -208,12 +209,178 @@ extension RunLoop {
+        case entry = 0
+        case beforeTimers = 1
+        case beforeSources = 2
+        case beforeWaiting = 32
+        case afterWaiting = 64
+        case exit = 128
+    }
+    
+    @available(*, deprecated, message: "For XCTest use only.")
+    public struct _Activities: OptionSet {
+        public var rawValue: UInt
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+        
+        public init(_ activity: _Activity) {
+            self.rawValue = activity.rawValue
+        }
+        
+        public static let entry = _Activities(rawValue: _Activity.entry.rawValue)
+        public static let beforeTimers = _Activities(rawValue: _Activity.beforeTimers.rawValue)
+        public static let beforeSources = _Activities(rawValue: _Activity.beforeSources.rawValue)
+        public static let beforeWaiting = _Activities(rawValue: _Activity.beforeWaiting.rawValue)
+        public static let afterWaiting = _Activities(rawValue: _Activity.afterWaiting.rawValue)
+        public static let exit = _Activities(rawValue: _Activity.exit.rawValue)
+        public static let allActivities = _Activities(rawValue: 0x0FFFFFFF)
+    }
+    
+    @available(*, deprecated, message: "For XCTest use only.")
+    public class _Observer {
+        fileprivate let cfObserver: CFRunLoopObserver
+        
+        fileprivate init(activities: _Activities, repeats: Bool, order: Int, handler: @escaping (_Activity) -> Void) {
+            self.cfObserver = CFRunLoopObserverCreateWithHandler(kCFAllocatorSystemDefault, CFOptionFlags(activities.rawValue), repeats, CFIndex(order), { (cfObserver, cfActivity) in
+                guard let activity = _Activity(rawValue: UInt(cfActivity.rawValue)) else { return }
+                handler(activity)
+            })
+        }
+        
+        public func invalidate() {
+            CFRunLoopObserverInvalidate(cfObserver)
+        }
+        
+        public var order: Int {
+            Int(CFRunLoopObserverGetOrder(cfObserver))
+        }
+        
+        public var isValid: Bool {
+            CFRunLoopObserverIsValid(cfObserver)
+        }
+        
+        deinit {
+            invalidate()
+        }
+    }
+    
+    @available(*, deprecated, message: "For XCTest use only.")
+    open class _Source: NSObject {
+        fileprivate var cfSource: CFRunLoopSource!
+        
+        public init(order: Int = 0) {
+            super.init()
+            
+            var context = CFRunLoopSourceContext(
+                version: 0,
+                info: Unmanaged.passUnretained(self).toOpaque(),
+                retain: nil,
+                release: nil,
+                copyDescription: { (info) -> Unmanaged<CFString>? in
+                    let me = Unmanaged<_Source>.fromOpaque(info!).takeUnretainedValue()
+                    return .passRetained(String(describing: me)._cfObject)
+                },
+                equal: { (infoA, infoB) -> DarwinBoolean in
```

Comment:
The `DarwinBoolean` is not portable, can we not use the `DarwinBoolean` -> `Bool` bridging here?

Reply:
This is a case where the types are different between platforms and I just need to drop the return value there, I think.

Reply:
File this under: why we want to stop exposing CF>